pkg/controller/perconaservermongodb/smart.go

```diff
@@ -313,11 +321,6 @@ func (r *ReconcilePerconaServerMongoDB) smartMongosUpdate(ctx context.Context, c
 
 	log.Info("StatefulSet is changed, starting smart update", "name", sts.Name)
 
-	if sts.Status.ReadyReplicas < sts.Status.Replicas {
-		log.Info("can't start/continue 'SmartUpdate': waiting for all replicas are ready")
-		return nil
-	}
-
```

Comment:
As I remember, during the certificate update, some mongos pods were not in the ready state. To fix this, the pods should be updated with new certificates, using `applyNWait` down below. But we couldn't reach this method because of this check